cmd/queue/main.go

```diff
@@ -201,8 +205,55 @@ func createAdminHandlers() *http.ServeMux {
 	return mux
 }
 
+func probeQueueHealthPath(port int) error {
+	url := fmt.Sprintf("http://127.0.0.1:%d%s", port, queue.RequestQueueHealthPath)
+
+	httpClient := &http.Client{
+		Transport: &http.Transport{
+			// Do not use the cached connection
+			DisableKeepAlives: true,
+		},
+		Timeout: 500 * time.Millisecond,
+	}
+
+	var lastErr error
+
+	timeoutErr := wait.PollImmediate(100*time.Millisecond, time.Second, func() (bool, error) {
```

Comment:
Should we give this some more leeway? As it stands, this will fail if the queue-proxy doesn't actually come up for a second. It feels to me as if we could have a much tighter retry interval because the actual request will again block as soon as the queue-proxy comes up until the user-container was successfully probed.

As such, each retry is pretty cheap and we can likely this to 50ms or even lower. However, the upper-bound of this can I think be bumped towards 5 or even 10 seconds to make sure we practically never retry on kube's `periodSeconds` interval.

Reply:
+1 - if we long poll here we can make this initial retry loop very tight.

Can we also pull this loop time out in to a constant?

Reply:
Turned this up to a 10 second timeout with a 25 millisecond interval.

We should totally take advantage of the full 10 second timeout on the ExecProbe. That was an oversight in the initial implementation.

The 25ms retry interval is a totally made up number. 🤷‍♂️

Reply:
25ms (40 times/sec) seems like it might be too aggressive. Help me convince myself that's not true.

Up to the provided frequency, we will:
1. Open a TCP socket
2. Send an HTTP request
3. Get an HTTP response
4. Check that the response code is 200

But, `wait.PollImmediate` will ignore timer ticks while the ConditionFunc is being evaluated, so if it takes (for example) 33ms to do 1-4, you will end up polling every 50ms. Additionally, this will only happen until we get a 200, so in the normal case, we don't care about the retry window.

The worst case for this would be something that actually spends a bunch of time working to get a good health check answer, and these requests end up clogging up the CPU, while blocking _actual_ initialization from taking place (which would end the health check pounding).

Might be worth documenting next to the 25ms that the number was made up as a balance between "don't make the user wait" and "don't pummel the container too hard".

Reply:
Added a comment explaining the "best guess" nature of the retry interval. Seems like something to keep an eye on and adjust in future.

Reply:
Adding some detail for the sake of clarity:

In the queue-proxy we probing here once a connection is able to be established we should hold it open for up to 10s. This means there are two cases for work we will do here that are a bit different than mentioned:

until queue-proxy is running we will:

1. open a tcp socket
2. get a RST (since the socket is not yet listening) and retry

This is the loop we want to optimize to retry quickly as it is very lightweight. Once queue proxy is up we do essentially as you describe (send a http req, get response), but waiting on the response will block up to 10s until the user container is ready so our poll frequency here will be mostly irrelevent.